GLMakie/src/drawing_primitives.jl

```diff
@@ -221,6 +221,32 @@ function get_space(x)
     return haskey(x, :markerspace) ? x.markerspace : x.space
 end
 
+function apply_transform_and_f32_conversion(
+        scene::Scene, plot::AbstractPlot, data,
+        space::Observable = get(plot, :space, Observable(:data))
+    )
+    f32c_obs = isnothing(scene.float32convert) ? Observable(nothing) : scene.float32convert.scaling
+    return map(plot, f32c_obs, transform_func_obs(plot), data, space) do _, _tf, data, space
+        tf = space == :data ? _tf : identity
+        f32c = space in (:data, :transformed) ? scene.float32convert : nothing
+        # avoid intermediate array?
+        return [Makie.f32_convert(f32c, apply_transform(tf, x)) for x in data]
+    end
+end
+
+# For Vector{<: Real} applying to x/y/z dimension
+function apply_transform_and_f32_conversion(
+        scene::Scene, plot::AbstractPlot, data, dim::Integer,
+        space::Observable = get(plot, :space, Observable(:data))
+    )
+    f32c_obs = isnothing(scene.float32convert) ? Observable(nothing) : scene.float32convert.scaling
+    return map(plot, f32c_obs, transform_func_obs(plot), data, space) do _, _tf, data, space
+        tf = space == :data ? _tf : identity
+        f32c = space in (:data, :transformed) ? scene.float32convert : nothing
+        return [Makie.f32_convert(f32c, apply_transform(tf, x), dim) for x in data]
```

Comment:
It's identity in that case

Reply:
Ah I somehow missed that line 😅 - thanks!

Reply:
Something I'm not too sure about here is whether it's ok to apply `transform_func` per element. It would be good for performance I assume, since we save an intermediate array...